pkg/utils/math_utils.go

```diff
@@ -63,3 +63,42 @@ func SafeDivBig(x *big.Int, y *big.Int) (*big.Int, error) {
 	}
 	return q, nil
 }
+
+// Finds a nonnegative integer x < p such that (m * x) % p == n.
+func DivMod(n *big.Int, m *big.Int, p *big.Int) (*big.Int, error) {
```

Comment:
did not this function already exists in `math_utils/utils.go` ? why do we have two implementations?

Reply:
yes, Ill remove the redundant implementation and merge both packages